lib/wallet.spec.js

```diff
@@ -61,149 +62,180 @@ function proxyquireWallet() {
 }
 
 
-describe('Wallet', () => {
-    let wallet;
-
-    beforeEach(() => {
-        stubbedProvider.getApiAccessToken.resolves(apiAccessToken);
-        stubbedProvider.getSupportedTokens.resolves(testTokens);
-        const Wallet = proxyquireWallet();
-        wallet = new Wallet(privateKey, stubbedProvider);
-    });
-
-    afterEach(() => {
-        stubbedErc20Contract.approve.reset();
-        stubbedClientFundContract.depositTokens.reset();
-        stubbedClientFundContract.withdraw.reset();
-        stubbedClientFundContract.unstage.reset();
-        stubbedProvider.getNahmiiBalances.reset();
-        stubbedProvider.getSupportedTokens.reset();
-        stubbedProvider.getTransactionConfirmation.reset();
-        stubbedProvider.getApiAccessToken.reset();
-    });
+[
+    privateKey,
+    {
+        address: '0x691A8D05678FC962ff0f2174134379c0051Cb686',
+        signMessage: sinon.stub(),
+        signTransaction: sinon.stub()
+    }
+].forEach(signer => {
+    describe('Wallet', () => {
+        let wallet;
 
-    context('an empty wallet', () => {
         beforeEach(() => {
-            stubbedProvider.getNahmiiBalances
-                .withArgs(walletAddress)
-                .resolves([]);
+            stubbedProvider.getApiAccessToken.resolves(apiAccessToken);
+            stubbedProvider.getSupportedTokens.resolves(testTokens);
+            const Wallet = proxyquireWallet();
+            wallet = new Wallet(privateKey, stubbedProvider);
```

Comment:
This seems strange.. you are using "privateKey" here, not "signer", then you override it with another beforeEach() below

Reply:
Good pickup